python/dify_plugin/invocations/storage.py

```diff
@@ -61,4 +90,4 @@ def exist(self, key: str) -> bool:
         ):
             return data["data"]
 
-        raise Exception("no data found")
\ No newline at end of file
+        raise NotFoundError("no data found")
```

Comment:
I'm uncertain whether `NotFoundError` is the appropriate exception to raise in this context.

If the key does not exist, I believe the `exist` method should return `false` as per its semantic meaning. 

However, if the generator yields no `data`, it might be more appropriate to raise a `StorageInvocationError`. This would signal a potential issue with the `Storage` implementation, providing clearer insight into what might be going wrong.